src/lib.rs

```diff
@@ -992,6 +1011,15 @@ impl<TRepoTypes: RepoTypes> Future for IpfsFuture<TRepoTypes> {
                             .collect();
                         let _ = ret.send(peers);
                     }
+                    IpfsEvent::FindPeer(peer_id, ret) => {
+                        let known_addrs = self.swarm.swarm.addresses_of_peer(&peer_id);
+                        let addrs = if !known_addrs.is_empty() {
+                            known_addrs
+                        } else {
+                            self.swarm.kademlia().addresses_of_peer(&peer_id)
+                        };
+                        let _ = ret.send(addrs);
```

Comment:
should there be the fallback dht search?

Reply:
yes, and there will be in a follow up; for now only the local DHT records are searched - I'll update the PR message to make this clear

Reply:
Yeah, now I am good to merge this :)